tests/test_native_mcm.py

```diff
@@ -0,0 +1,424 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for default qubit preprocessing."""
+from functools import reduce
+from typing import Sequence
+
+import numpy as np
+import pennylane as qml
+import pytest
+from conftest import LightningDevice as ld
+from conftest import device_name
+from flaky import flaky
+from pennylane._device import DeviceError
+
+if not ld._CPP_BINARY_AVAILABLE:
+    pytest.skip("No binary module found. Skipping.", allow_module_level=True)
+
+if device_name != "lightning.qubit":
+    pytest.skip(f"Native MCM not supported in device {device_name}.", allow_module_level=True)
```

Comment:
```suggestion
if not LightningDevice._new_API:
    pytest.skip("Exclusive tests for new device API. Skipping.", allow_module_level=True)

if not LightningDevice._CPP_BINARY_AVAILABLE:  # pylint: disable=protected-access
    pytest.skip("No binary module found. Skipping.", allow_module_level=True)
```
Let's keep it in a standard way with the other tests in this folder. My idea with this is to make it as straightforward as possible for someone to migrate our other devices.